enterprise/models.py

```diff
@@ -270,6 +270,37 @@ def user_email(self):
             return self.user.email
         return None
 
+    @property
+    def entitlements(self):
+        """
+        Return entitlement ids available to the learner.
+
+        Returns an empty list if enterprise customer requires data sharing consent and learner has not agreed.
+
+        Returns:
+            (list): A list of entitlements that learner can avail.
+        """
+        # Check if Enterprise Customer requires data sharing consent and store the boolean result
+        # pylint: disable=invalid-name
+        enterprise_customer_requires_consent = \
+            self.enterprise_customer.enforces_data_sharing_consent(EnterpriseCustomer.AT_LOGIN) or \
+            self.enterprise_customer.enforces_data_sharing_consent(EnterpriseCustomer.AT_ENROLLMENT)
+
+        # Check if Enterprise Learner consents to data sharing and store the boolean result
+        learner_consent_state = self.data_sharing_consent.first()
```

Comment:
`data_sharing_consent` comes via a reverse relationship on `UserDataSharingConsentAudit ` [here](https://github.com/edx/edx-enterprise/blob/master/enterprise/models.py#L519). Technically, this is a OneToMany relationship but our code and current business requirements make sure that it is OneToOne relationship.

to answer your original question, value of `learner_consent_state ` is an instance of `UserDataSharingConsentAudit` model.